presto-main/src/main/java/com/facebook/presto/sql/planner/TranslationMap.java

```diff
@@ -270,6 +284,22 @@ private Expression coerceIfNecessary(Expression original, Expression rewritten)
                 }
                 return rewritten;
             }
+
+            private Expression rewriteEnumLiteral(DereferenceExpression original, EnumType enumType)
+            {
+                Object enumValue = resolveEnumLiteral(original, enumType);
+                if (enumType instanceof LongEnumType) {
+                    return new Cast(
+                            new LongLiteral(((Long) enumValue).toString()),
+                            enumType.getTypeSignature().getBase());
+                }
+                if (enumType instanceof VarcharEnumType) {
+                    return new Cast(
+                            new StringLiteral(((Slice) enumValue).toStringUtf8()),
+                            enumType.getTypeSignature().getBase());
+                }
+                throw new IllegalArgumentException("Unsupported enum type");
```

Comment:
I think you should just have a `checkArgument(enumType instanceof LongEnumType || enumType instanceof VarcharEnumType, format("Unsupported enumType '%s (%s)'", enumType, enumType.class))` at the beginning of the function. We normally do validation first.